### PR TITLE
supply.html & wave-generator.html modified

### PR DESCRIPTION
**Pull request by frostcaltrop**

### CODE REVIEW

File: wwwroot/oscilloscope.html [hunk 1]
```diff
@@ -719,5 +719,8 @@ <h1 class="mdui-typo-title" i18n-key="osc-trigger-attribute"></h1>
     i18n.LocalizeFile();
 </script>
 <script src="js/mdui.min.js" type="text/javascript"></script>
+    <script type="text/javascript">
```
Comment (84634E1A607A) [inline wwwroot/oscilloscope.html:722]: Have a look at line 719...

File: wwwroot/js/i18n.js [hunk 2]
```diff
@@ -101,6 +139,44 @@ class i18n {
...
+        "wave-type-dc": "DC",
+        "wave-type-sine": "Sine",
+        "wave-type-square": "Square",
+        "wave-type-triangle": "Triangle",
+        "wave-type-ramp-up": "RampUp",
+        "wave-type-ramp-down": "RampDown",
+        "wave-type-noise": "Noise",
+        "wave-type-pulse": "Pulse",
+        "wave-type-trapezium": "Trapezium",
+        "wave-type-sinepower": "SinePower",
```
Comment (84634E1A607A) [inline wwwroot/js/i18n.js:167]: I'd add a space between the words. e.g. `Ramp Up`

File: wwwroot/js/i18n.js [hunk 3]
```diff
@@ -49,6 +49,44 @@ class i18n {
...
+        "wave-msg-invalid-frequency": "请输入正确的频率。",
+        "wave-msg-too-high-frequency": "{EngineeringValue.To(value)}Hz 频率过高。",
+        "wave-msg-not-positive-frequency": "{EngineeringValue.To(value)}Hz 应为正数。",
+        "wave-msg-invalid-amp": "请输入正确的幅值。",
+        "wave-msg-too-high-amp": "{EngineeringValue.To(value)}V 幅值过高。",
+        "wave-msg-too-low-amp": "{EngineeringValue.To(value)}V 幅值过低。",
+        "wave-msg-invalid-offset": "请输入正确的偏置。",
+        "wave-msg-too-high-offset": "{EngineeringValue.To(value)}V 偏置过高。",
+        "wave-msg-too-low-offset": "{EngineeringValue.To(value)}V 偏置过低。",
+        "wave-msg-invalid-svmme": "请输入正确的对称度。",
```
Comment (84634E1A607A) [inline wwwroot/js/i18n.js:87]: What's svmme? "Symmetry" is sym...

File: wwwroot/supply.html [hunk 4]
```diff
@@ -33,18 +33,18 @@
...
 <div class="mdui-drawer" id="main-drawer">
     <div class="mdui-list">
-        <a class="mdui-list-item mdui-ripple" href="/Index">Home</a>
-        <a class="mdui-list-item mdui-ripple" href="/Oscilloscope">Oscilloscope</a>
-        <a class="mdui-list-item mdui-ripple" href="/Supply">Power Source</a>
-        <a class="mdui-list-item mdui-ripple" href="wave-generator.html">Wave Gen</a>
+        <a class="mdui-list-item mdui-ripple" href="/Index" i18n-key="site-home"></a>
+        <a class="mdui-list-item mdui-ripple" href="/Oscilloscope" i18n-key="site-osc"></a>
+        <a class="mdui-list-item mdui-ripple" href="/Supply" i18n-key="site-supply"></a>
+        <a class="mdui-list-item mdui-ripple" href="wave-generator.html" i18n-key="site-wave"></a>
```
Comment (84634E1A607A) [inline wwwroot/supply.html:39]: Remind me of this leftover issue.

File: wwwroot/supply.html [hunk 5]
```diff
@@ -91,21 +91,21 @@ <h1 class="mdui-typo-title">负通道</h1>
...
-                    message: `${EngineeringValue.To(value)}V is too high.`,
+                    message: i18n.Localize("supply-msg-too-high-vol").replace("{EngineeringValue.To(value)}", EngineeringValue.To(value)),
                     position: "right-top"
                 });
                 return;
             }
             if ((isNeg ? -1 : 1) * value < 0) {
                 mdui.snackbar({
-                    message: `${EngineeringValue.To(value)}V should be ${isNeg ? "negative" : "positive"}.`,
+                    message: i18n.Localize("supply-msg-false-pos-neg-set").replace("{EngineeringValue.To(value)}", EngineeringValue.To(value)).replace("{isNeg ? \"negative\" : \"positive\"}", isNeg ? "negative" : "positive"),
```
Comment (84634E1A607A) [inline wwwroot/supply.html:108]: I'll take care of this.

File: wwwroot/wave-generator.html [hunk 6]
```diff
@@ -11,402 +11,403 @@
...
+                }
             }
-        }
-    })
-</script>
-<script src="js/mdui.min.js" type="text/javascript"></script>
+        })
+    </script>
+    <script src="js/mdui.min.js" type="text/javascript"></script>
+    <script type="text/javascript">i18n.LocalizeFile()</script>
```
Comment (84634E1A607A) [inline wwwroot/wave-generator.html:411]: Line 385

File: wwwroot/wave-generator.html [hunk 7]
```diff
@@ -11,402 +11,403 @@
...
-            <div class="mdui-col-xs-11">
-                <h1 class="mdui-typo-title">通道${ch}</h1>
-            </div>
-            <div class="mdui-col-xs-1" style="align-self: center; text-align: right">
-                <label class="mdui-switch">
-                    <input class="mdui-switch" id="wave-gen-${ch}-enable" type="checkbox"/>
-                    <i class="mdui-switch-icon"></i>
-                </label>
-            </div>
+    <header class="mdui-appbar mdui-appbar-fixed">
```
Comment (84634E1A607A) [inline wwwroot/wave-generator.html:14]: The massive format changes here make it very difficult to determine whether the change is good or not. I'll take close look on it when available.

**Review by 84634E1A607A — CHANGES_REQUESTED** [review]

File: wwwroot/js/i18n.js (hunk 3 above)
Comment (frostcaltrop) [reply wwwroot/js/i18n.js:86]: my fault

Comment (frostcaltrop) [reply wwwroot/js/i18n.js:86]: my fault

File: wwwroot/oscilloscope.html (hunk 1 above)
Comment (frostcaltrop) [reply wwwroot/oscilloscope.html:722]: oh I forgot

**Comment by 84634E1A607A** [comment]
Will make one myself

**Pull request closed without merging**